content/02-types.md

```diff
@@ -870,6 +870,27 @@ One problem may be apparent - what happens if a member function is not declared
 >
 > Before the advent of abstract types, all basic types were implemented as extern classes or enums. While this nicely took care of some aspects such as `Int` being a "child class" of `Float`, it caused issues elsewhere. For instance, with `Float` being an extern class, it would unify with the empty structure `{}`, making it impossible to constrain a type to accept only real objects.
 
+<!--label:types-abstract-this-->
+#### Access to Underlying Data
+
+As shown in the example from the previous section, `this` in abstract methods refers to the underlying data. For `AbstractInt` methods, `this` is therefore a variable of type `Int`.
+
+[code asset](assets/MyAbstract.hx#L1-L5)
+
+##### since Haxe 4.3.0
+
+The `abstract` keyword can be used in abstract methods to refer to the current instance *as an abstract type*, rather than referring to the underlying data with the `this` keyword.
+
+This can be useful when an abstract method needs to call other methods which accept an argument of the abstract type. For example, suppose class `Main` defines a method `takeAbstractInt`:
```

Comment:
> which accept an argument of the abstract type

I'm having some trouble understanding what that means exactly. How do arguments come into play here? The important bit should be that the code in an abstract method wants to call a method "of the abstract".

Reply:
Ok reading the next part (and looking at the code) clarifies this somewhat. I feel like this focuses on this specific case a little too much, but it's good to get the difference across.

Reply:
That would work too… The example here is just a static method which takes a `AbstractInt` instance. But it would also work if I wanted to call something like `abstract.anotherInstanceMethod()`.

Reply:
A switch case for enum abstracts might also be another good example of when this comes in useful.
https://github.com/HaxeFoundation/haxe/wiki/Breaking-changes-in-Haxe-4.0.0#switching-on-this-in-member-methods-of-enum-abstracts

It was also mentioned in the original proposal.